overlord/devicestate/handlers.go

```diff
@@ -926,13 +926,9 @@ func gadgetCurrentAndUpdate(st *state.State, snapsup *snapstate.SnapSetup) (curr
 }
 
 var (
-	gadgetUpdate = nopGadgetOp
+	gadgetUpdate = gadget.Update
```

Comment:
can't we have a minimal unit test about this change?

Reply:
I'll add one